graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -79,112 +70,17 @@ def _convert_uuid_string_to_int(uuid_string):
-        lower_bound_b: int or None, describing one of the lower bounds. If the bound does not exist,
-                       the argument value should be None.
-
-    Returns:
-        - int, the larger of the two lower bounds, if one or more lower bounds have an integer
-          value.
-        - None if both lower bounds have a value of None.
-    """
-    stronger_lower_bound = None
-    if lower_bound_a is not None and lower_bound_b is not None:
-        stronger_lower_bound = max(lower_bound_a, lower_bound_b)
-    elif lower_bound_a is not None:
-        stronger_lower_bound = lower_bound_a
-    elif lower_bound_b is not None:
-        stronger_lower_bound = lower_bound_b
-
-    return stronger_lower_bound
-
-
-def _get_stronger_upper_bound(upper_bound_a, upper_bound_b):
-    """Return the smaller bound of the two given upper bounds.
-
-    Args:
-        upper_bound_a: int or None, describing one of the upper bounds. If the bound does not exist,
-                       the argument value should be None.
-        upper_bound_b: int or None, describing one of the upper bounds. If the bound does not exist,
-                       the argument value should be None.
-
-    Returns:
-        - int, the smaller of the two upper bounds, if one or more upper bounds have an integer
-          value.
-        - None if both upper bounds have a value of None.
-    """
-    stronger_upper_bound = None
-    if upper_bound_a is not None and upper_bound_b is not None:
-        stronger_upper_bound = min(upper_bound_a, upper_bound_b)
-    elif upper_bound_a is not None:
-        stronger_upper_bound = upper_bound_a
-    elif upper_bound_b is not None:
-        stronger_upper_bound = upper_bound_b
-
-    return stronger_upper_bound
-
-
-def _get_intersection_of_intervals(interval_a, interval_b):
-    """Return the intersection of two IntegerIntervals, or None if the intervals are disjoint.
-
-    Args:
-        interval_a: None or IntegerInterval namedtuple. None means empty interval.
-        interval_b: None or IntegerInterval namedtuple. None means empty interval.
-
-    Returns:
-        - IntegerInterval namedtuple, intersection of the two given IntegerIntervals, if the
-          intersection is not empty.
-        - None otherwise.
-    """
-    if interval_a is None or interval_b is None:
-        return None
-
-    strong_lower_bound = _get_stronger_lower_bound(interval_a.lower_bound, interval_b.lower_bound)
-    strong_upper_bound = _get_stronger_upper_bound(interval_a.upper_bound, interval_b.upper_bound)
-
-    intersection = _create_integer_interval(strong_lower_bound, strong_upper_bound)
-    return intersection
-
-
-def _get_query_interval_of_binary_integer_inequality_filter(parameter_values, filter_operator):
-    """Return IntegerInterval or None of values passing through a binary integer inequality filter.
+def _get_query_interval_of_binary_integer_inequality_filter(
+    parameter_values: List[int], filter_operator: str
```

Comment:
Maybe `Collection[int]` rather than `List[int]`? I'm not sure the list-iness matters here.

Reply:
Maybe it doesn't. Is it worth checking? Mypy is optional type hints for a reason, let's not obsess about details.